custom/icds_reports/tasks.py

```diff
@@ -44,27 +49,65 @@ def move_ucr_data_into_aggregation_tables(date=None, intervals=3):
                 cursor.execute(sql_to_execute)
             celery_task_logger.info("Ended icds reports update_location_tables_sql")
 
-            path = os.path.join(os.path.dirname(__file__), 'sql_templates', 'update_monthly_aggregate_tables.sql')
+        aggregation_tasks = []
+
+        for interval in range(intervals - 1, -1, -1):
+            calculation_date = (monthly_date - relativedelta(months=interval)).strftime('%Y-%m-%d')
+            aggregation_tasks.append(UCRAggregationTask('monthly', calculation_date))
+
+        aggregation_tasks.append(UCRAggregationTask('daily', date.strftime('%Y-%m-%d')))
+        aggregate_tables.delay(aggregation_tasks[0], aggregation_tasks[1:])
```

Comment:
Is there a concern of aggregate data over daily vs monthly being inconsistent if one task fails but other succeeds?

Reply:
If it fails then https://github.com/dimagi/commcare-hq/pull/17749/files#diff-b8c5a5e308a203bcaee8ed83e6837edfR96 will raise an exception and it won't queue the following tasks, but it will retry the current tasks, so there should never be a time when one is queued and the previous one did not finish